lib/amazon_mws.rb

```diff
@@ -12,7 +12,7 @@
 
 class AmazonMwsOrderResponse
   def initialize(response)
-    @response = response.fetch("GetOrderReferenceDetailsResponse", {})
+    @response = Hash.from_xml(response.body).fetch("GetOrderReferenceDetailsResponse", {})
```

Comment:
I would have to create a class that inherits SpreeAmazon::Response called GetOrderReferenceDetails and do some customization since the GetOrderReferenceDetails response structure is different than other api calls